Intersect.Server/Maps/MapController.cs

```diff
@@ -0,0 +1,540 @@
+                {
+                    mSurroundingMapIds = value;
+                    var surroundingMapsIdsWithSelf = new List<Guid>(value);
+                    surroundingMapsIdsWithSelf.Add(Id);
+                    mSurroundingMapsIdsWithSelf = surroundingMapsIdsWithSelf.ToArray();
+                }
+            }
+        }
+
+        [JsonIgnore]
+        [NotMapped]
+        public MapController[] SurroundingMaps
+        {
+            get => mSurroundingMaps;
+
+            set
+            {
+                lock (GetMapLock())
+                {
+                    mSurroundingMaps = value;
+                    var surroundingMapsWithSelf = new List<MapController>(value);
+                    surroundingMapsWithSelf.Add(this);
+                    mSurroundingMapsWithSelf = surroundingMapsWithSelf.ToArray();
+                }
+            }
+        }
+
+        //EF
+        public MapController() : base()
+        {
+            Name = "New Map";
+            Layers = null;
+        }
+
+        //For New Maps!
+        public MapController(bool newMap = false) : base(Guid.NewGuid())
+        {
+            Name = "New Map";
+            Layers = null;
+        }
+
+        [JsonConstructor]
+        public MapController(Guid id) : base(id)
+        {
+            if (id == Guid.Empty)
+            {
+                return;
+            }
+
+            Layers = null;
+        }
+
+        public new static MapControllers Lookup => sLookup = sLookup ?? new MapControllers(MapBase.Lookup);
+
+        public static MapController Get(Guid id)
+        {
+            return Lookup.Get<MapController>(id);
+        }
+
+        public static bool TryGetInstanceFromMap(Guid mapControllerId, Guid instanceId, out MapInstance mapInstance, bool createIfNew = false)
+        {
+            mapInstance = null;
+            var controller = Get(mapControllerId);
+            if (controller != null && controller.TryGetInstance(instanceId, out mapInstance, createIfNew))
+            {
+                return mapInstance != null;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public static List<MapInstance> GetSurroundingMapInstances(Guid mapControllerId, Guid instanceId, bool includeSelf, bool createIfNew = false)
+        {
+            List<MapInstance> instances = new List<MapInstance>();
+            var controller = Get(mapControllerId);
+            if (controller != null)
+            {
+                foreach (var controllerId in controller.GetSurroundingMapIds(includeSelf).ToList())
+                {
+                    if (TryGetInstanceFromMap(controllerId, instanceId, out var instance))
+                    {
+                        instances.Add(instance);
+                    }
+                }
+            }
+            return instances;
+        }
+
+        //GameObject Functions
+
+        public object GetMapLock()
```

Comment:
Documentation?
(I think you get the idea for the rest of the file.. there's a lot of methods :P)